src/components/TimeBasedLineChart.js

```diff
@@ -0,0 +1,753 @@
+const React = require('react');
+const ReactDom = require('react-dom');
+const Radium = require('radium');
+
+const d3 = require('d3');
+const moment = require('moment');
+const numeral = require('numeral');
+
+const StyleConstants = require('../constants/Style');
+
+class TimeBasedLineChart extends React.Component {
+  constructor (props) {
+    super(props);
+
+    const adjustedWidth = this.props.width - this.props.margin.right - this.props.margin.left;
+    const adjustedHeight = this.props.height - this.props.margin.top - this.props.margin.bottom;
+
+    this.state = {
+      adjustedHeight,
+      adjustedWidth
+    };
+  }
+
+  componentDidMount () {
+    this.setState({
+      chartEl: ReactDom.findDOMNode(this.refs.chart)
+    });
+  }
+
+  componentWillReceiveProps (newProps) {
+    if (newProps.height !== null || newProps.width !== null || newProps.margin !== null) {
+      const height = newProps.height || this.props.height;
+      const width = newProps.width || this.props.width;
+      const margin = newProps.margin || this.props.margin;
+
+      const adjustedWidth = width - margin.right - margin.left;
+      const adjustedHeight = height - margin.top - margin.bottom;
+
+      this.setState({
+        adjustedHeight,
+        adjustedWidth
+      });
+    }
+  }
+
+  _getAreaBelowZero (data) {
+    const area = d3.svg.area()
+      .x(d => {
+        const currentDate = moment.unix(d.timeStamp).startOf(this.props.rangeType).unix()
+
+        return this._getXScaleValue(currentDate);
+      })
+      .y0(this.state.adjustedHeight - 10)
+      .y1(this._getYScaleValue(0));
+
+    return area(data);
+  }
+
+  _getFlatLine (data) {
```

Comment:
No, I was going to use it for animations but haven't gotten that far.  I'll remove it for now.